Modelica/Clocked/ClockSignals.mo

```diff
@@ -0,0 +1,1227 @@
+
+      parameter Integer factor(min=0)
+        "Sample factor with respect to resolution"                               annotation(Evaluate=true);
+      parameter Clocked.Types.Resolution resolution=R.ms "Clock resolution"
+        annotation (Evaluate=true, __Dymola_editText=false);
+      extends Clocked.ClockSignals.Interfaces.PartialPeriodicClock;
+    protected
+      constant Integer conversionTable[8]={365*24*60*60, 24*60*60, 60*60, 60, 1, 1000, 1000*1000, 1000*1000*1000}
+        "Table to convert from Resolution to Integer clock resolution";
+      parameter Integer resolutionFactor = conversionTable[Integer(resolution)] annotation(Evaluate=true);
+      Clock c annotation(HideResult=true);
+    equation
+      // The following then-branch using subSample corresponds to the simpler Clock(factor*resolutionFactor, 1),
+      // but works better in some odd cases.
+      //
+      // Specifically the simple variant may fail if the product exceeds 2^31 (seconds), roughly 68 years.
+      //
+      // Using subSample works reliably up to 2^31 years according to the standard.
+      //
+      // There is no similar issue with the else-branch.
+      if resolution < R.s then
+         c = subSample(Clock(factor), resolutionFactor);
+      else
+         c = Clock(factor, resolutionFactor);
+      end if;
+
+      if useSolver then
+         y = Clock(c, solverMethod=solverMethod);
+      else
+         y = c;
+      end if;
+
+      annotation (
+           defaultComponentName="periodicClock1",
+        Icon(coordinateSystem(preserveAspectRatio=true,  extent={{-100,-100},{100,100}}, initialScale=0.06),
+                        graphics={
+            Text(
+              extent={{-200,-110},{200,-175}},
+              textString="%factor %resolution"),
+            Rectangle(
+              extent={{20,58},{40,46}},
+              fillPattern=FillPattern.Solid,
+              rotation=90,
+              origin={52,60},
+              lineColor={95,95,95},
+              fillColor={95,95,95}),
+            Rectangle(
+              extent={{20,58},{40,46}},
+              fillPattern=FillPattern.Solid,
+              rotation=45,
+              origin={-50,-120},
+              lineColor={95,95,95},
+              fillColor={95,95,95}),
+            Text(
+              visible=useSolver,
+              extent={{-300,-180},{300,-245}},
+              textString="%solverMethod")}),
+        Diagram,
+        Documentation(info="<html>
+<p>
+This component generates a periodic clock that starts ticking when
+the simulation starts. The <b>period</b> is defined as the product
+of a resolution, defined with enumeration
+<a href=\"modelica://Modelica.Clocked.Types.Resolution\">Resolution</a>,
+and Integer parameter <b>factor</b>. Internally, the period is
+represented as a rational number. All clocks with rational number definitions
+are exactly time synchronized to each other.
+</p>
+
+<p>
+Example:
+</p>
+
+<pre>
+  import M = Modelica.Clocked;
```

Comment:
The renaming (and shortening) import is no longer endorsed (even though it is used in the specification a lot). Better use `import Modelica.Clocked;`.